python/ray/data/_internal/execution/interfaces/execution_options.py

```diff
@@ -105,6 +108,10 @@ class ExecutionOptions:
 
     resource_limits: ExecutionResources = field(default_factory=ExecutionResources)
 
+    reserved_resources: ExecutionResources = field(
```

Comment:
Can we add a check that this is not set if resource_limits is manually specified?